features/nanostack/nanostack-interface/Nanostack.cpp

```diff
@@ -648,6 +648,18 @@ nsapi_size_or_error_t Nanostack::do_sendto(void *handle, const ns_address_t *add
      * */
     if (retcode == NS_EWOULDBLOCK) {
         ret = NSAPI_ERROR_WOULD_BLOCK;
+    } else if (retcode == -1) {
+        ret = NSAPI_ERROR_PARAMETER;
```

Comment:
If we refer to Nanostack's socket API:
```c++
/**
 ...
 * Error returns:
 *
 * \return -1 Invalid socket ID.
 * \return -2 Socket memory allocation fail.
 * \return -3 TCP state not established or address scope not defined .
 * \return -4 Unknown interface.
 * \return -5 Socket not connected
 * \return -6 Packet too short (ICMP raw socket error).
 */
int16_t socket_sendmsg(int8_t socket, const ns_msghdr_t *msg, int flags);
```

then should -1 be `NSAPI_ERROR_NO_SOCKET`

Reply:
This is [how -1 is returned](https://github.com/ARMmbed/mbed-os/blob/93dc5514f2502409edf41fcd504efbec70ae2aa0/features/nanostack/sal-stack-nanostack/source/libNET/src/socket_api.c#L503):
```
    if (!msg) {
        return -1;
    }
```
Or [here](https://github.com/ARMmbed/mbed-os/blob/93dc5514f2502409edf41fcd504efbec70ae2aa0/features/nanostack/sal-stack-nanostack/source/Core/ns_socket.c#L1007) (I think this is the one they meant by "Invalid socket ID."):
```
    if (!socket_ptr || !socket_ptr->inet_pcb) {
        tr_warn("Socket Id %i", sid);
        ret_val = -1;
        goto fail;
    }
```
Or [here](https://github.com/ARMmbed/mbed-os/blob/93dc5514f2502409edf41fcd504efbec70ae2aa0/features/nanostack/sal-stack-nanostack/source/Core/ns_socket.c#L1016):
```
        if (!socket_message_validate_iov(msg, &payload_length)) {
            return -1;
        }
```
Looks to me like -1 is returned if the input parameters are wrong.

Reply:
OK. Then its clearly a parameter error.

Reply:
Yes, -1 is generally "parameter error" in Nanostack, for the few socket functions that have multiple error returns. But the socket ID isn't distinguished from any other faulty parameter.

I would have thought there would be scope for `NSAPI_ERROR_NO_SOCKET` being generated in the wrapper, at least.